packages/core/shared/stream-parts.ts

```diff
@@ -90,12 +174,47 @@ type StreamPartValueType = {
 export type StreamPartType =
   | ReturnType<typeof textStreamPart.parse>
   | ReturnType<typeof functionCallStreamPart.parse>
-  | ReturnType<typeof dataStreamPart.parse>;
+  | ReturnType<typeof dataStreamPart.parse>
+  | ReturnType<typeof errorStreamPart.parse>
+  | ReturnType<typeof assistantMessage.parse>
+  | ReturnType<typeof assistantControlData.parse>;
```

Comment:
I'll look into reducing this code duplication (also in other places) in future PRs.